src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonReconciliationService.kt

```diff
@@ -0,0 +1,80 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails
+
+import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomisprisoner.model.PrisonerProfileDetailsResponse
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.model.SyncPrisonerDomesticStatusResponse
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.model.SyncPrisonerNumberOfChildrenResponse
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails.ContactPersonProfileType.CHILD
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails.ContactPersonProfileType.MARITAL
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.profiledetails.ProfileDetailsNomisApiService
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.doApiCallWithRetries
+
+@Service
+class ContactPersonReconciliationService(
+  @Autowired private val nomisApi: ProfileDetailsNomisApiService,
+  @Autowired private val dpsApi: ContactPersonProfileDetailsDpsApiService,
+  @Autowired private val telemetryClient: TelemetryClient,
+) {
+
+  suspend fun checkPrisoner(prisonerNumber: String): String? {
```

Comment:
Just done the service for a single prisoner for now.

Coming soon:

- checking for nulls in either DPS or NOMIS
- full reconciliation with start endpoint
- error handling
- single prisoner endpoint